src/Symfony/Component/Validator/Command/DebugCommand.php

```diff
@@ -168,6 +171,14 @@ private function getPropertyData(ClassMetadataInterface $classMetadata, string $
                     'options' => $this->getConstraintOptions($constraint),
                 ];
             }
+            if (CascadingStrategy::CASCADE === $metadata->getCascadingStrategy()) {
+                $constraint = new Valid([], [], null, TraversalStrategy::IMPLICIT === $metadata->getTraversalStrategy());
+                $data[] = [
+                    'class' => Valid::class,
+                    'groups' => [$constraint::DEFAULT_GROUP],
+                    'options' => $this->getConstraintOptions($constraint),
+                ];
```

Comment:
Hey, @macintoshplus thanks for your this PR! It's working great! But are we sure only Valid constraints have the CascadingStrategy::CASCADE? For example, if you create custom constraints that extend the Valid constraint you gonna have this CascadingStrategy::CASCADE, but you don't want to add here the Valid constraint but the custom one.

Reply:
It's true, if you extend `Valid` annotation without changing anything, the behavior is the same as using the `Valid` constraint. And, your custom constraint is changed to `Valid` on debug command. The debug command cannot know the original constraint class name because Symfony has removed it.

See the code https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Validator/Mapping/GenericMetadata.php#L145-L155

But if you add some options or groups, the original constraint is kept by Symfony and can be displayed on the dump command output.

Symfony change property mapping configuration with `EnableAutoMapping` and `DisableAutoMapping`.

Reply:
@macintoshplus Maybe you can check if the metadata is an instance of ClassMetadata and then getAutoMappingStragegy

Reply:
@loic425 Yes, I can.

During the investigation (thank @HeahDude), I found some other constraints with a particular behavior.

Without trying to retrieve the original constraint to display, it's possible to display class or property metadata properties differently? How?